docs/rst/tutorials/pkcs11_tutorial.rst

```diff
@@ -0,0 +1,168 @@
+.. _vulcanexus_pkcs11_tutorial:
+
+ROS 2 Security improved with Hardware Security Modules (HSM) using Vulcanexus Core
+==================================================================================
+
+.. contents::
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+`ROS 2 Security <https://docs.ros.org/en/humble/Concepts/About-Security.html>`_ is implemented by means of the `DDS Security specification <https://www.omg.org/spec/DDS-SECURITY/About-DDS-SECURITY/>`_.
+Specifically, when using Vulcanexus, security is implemented through `Fast DDS security plugins <https://fast-dds.docs.eprosima.com/en/latest/fastdds/security/security.html#security>`_.
+The security files required by the aforementioned specification are encapsulated within the ROS 2 security enclaves.
+This approach allows to set different security policies for different nodes, processes, users, or devices by defining different enclaves with their corresponding permissions.
+However, ROS 2 currently requires that the private key used to encrypt and decrypt the enclave information is also stored within the enclave as plain text, which might pose a security vulnerability.
+Hardware Security Modules (HSM) can be used in order to mitigate this vulnerability.
+And `PKCS #11 standard <http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html>`_ provides an API to interact with these devices.
+Vulcanexus Core provides the required support for HSM devices so the private key, instead of being stored in the filesystem, it could be inside the HSM device.
```

Comment:
```suggestion
Such vulnerability can be mitigated through the use of Hardware Security Modules (HSM).
These modules provide a query API compliant with the `PKCS #11 standard <http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html>`_ for which Vulcanexus Core (through Fast DDS) provides the required support so the private keys, instead of being stored in the filesystem, could be kept inside HSM devices.
```